### PR TITLE
EPMLSTRCMW-265 refactor: Replacing UserRepository with UserService

### PR DESCRIPTION
**Pull request by OlegKoshelev**
**DESCRIPTION**

Use `UserService` in `AuthService` instead of `UserRepository`

**CHANGES**
- AuthService.scala, ServiceModule.scala, UserService.scala, AuthServiceTest.scala and UserServiceTest.scala have been updated.
- It allows to avoid a break of encapsulation and future problems during refactoring.

### CODE REVIEW

**Review by DruzhininPavel — APPROVED** [review]
LGTM

**Review by Nikolai-Isaev — APPROVED** [review]
Lgtm

File: services/src/main/scala/cromwell/pipeline/service/UserService.scala [hunk 1]
```diff
@@ -50,3 +55,8 @@ class UserService(userRepository: UserRepository)(implicit executionContext: Exe
       }
     } else Future.failed(new RuntimeException("new password incorrectly duplicated"))
 }
+
+object UserService {
+  def apply(userRepository: UserRepository)(implicit executionContext: ExecutionContext): UserService =
+    new UserService(userRepository)(executionContext)
+}
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/UserService.scala]: This change is ok by itself, but it will make code style in `UserService` inconsistent with other services. We should either change all services to have an `apply` method, or don't change this one. Since we already have a task EPMLSTRCMW-241 where we create `apply` method for all services, I suggest you to just revert this change

**Review by gominasian — APPROVED** [review]

File: services/src/main/scala/cromwell/pipeline/service/UserService.scala (hunk 1 above)
Comment (OlegKoshelev) [reply services/src/main/scala/cromwell/pipeline/service/UserService.scala]: @myazinn done

File: services/src/main/scala/cromwell/pipeline/service/ServiceModule.scala [hunk 2]
```diff
@@ -16,7 +16,7 @@ class ServiceModule(
   implicit executionContext: ExecutionContext
 ) {
   lazy val authService: AuthService =
-    new AuthService(datastorageModule.userRepository, authModule.authUtils)
+    new AuthService(new UserService(datastorageModule.userRepository), authModule.authUtils)
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/ServiceModule.scala]: Sorry for not noticing it earlier, but why do we create another instance of `UserService` here? Can't we use `userService` defined on the line below?

**Review by myazinn — CHANGES_REQUESTED** [review]

File: services/src/main/scala/cromwell/pipeline/service/ServiceModule.scala (hunk 2 above)
Comment (OlegKoshelev) [reply services/src/main/scala/cromwell/pipeline/service/ServiceModule.scala]: Thanks! I did not notice

Comment (OlegKoshelev) [reply services/src/main/scala/cromwell/pipeline/service/ServiceModule.scala]: @myazinn done

**Review by myazinn — APPROVED** [review]

**Review by Dmitrii-Shevchenko — APPROVED** [review]
